tests/Infrastructure/Process/FileCopierTest.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+namespace PhpTuf\ComposerStager\Tests\Infrastructure\Process;
+
+use PhpTuf\ComposerStager\Exception\LogicException;
+use PhpTuf\ComposerStager\Exception\ProcessFailedException;
+use PhpTuf\ComposerStager\Infrastructure\Process\FileCopier;
+use PhpTuf\ComposerStager\Infrastructure\Process\Runner\RsyncRunner;
+use PhpTuf\ComposerStager\Tests\TestCase;
+use Prophecy\Argument;
+
+/**
+ * @coversDefaultClass \PhpTuf\ComposerStager\Infrastructure\Process\FileCopier
+ * @covers ::__construct()
+ *
+ * @property \PhpTuf\ComposerStager\Infrastructure\Process\Runner\RsyncRunner|\Prophecy\Prophecy\ObjectProphecy rsync
+ */
+class FileCopierTest extends TestCase
+{
+    public function setUp(): void
+    {
+        $this->rsync = $this->prophesize(RsyncRunner::class);
```

Comment:
🤔 Is `$this->rsync` formally defined anywhere?

Reply:
What do you mean "formally defined"? It's documented as a `@property` on the test class--which is how I always do it in my tests, to reduce boilerplate.

Reply:
I meant `private $rsync` as a concrete class member. But if you're doing it with `@property`, that's fine too.